### PR TITLE
Universal Token Address Resolution

### PR DESCRIPTION
**Pull request by r-near**
## Problem
The existing token resolution system is split into two functions:
- `get_token_address`: converts NEAR tokens to foreign chain addresses
- `get_token_id`: converts foreign chain addresses to NEAR tokens

This creates several issues:
1. No direct way to resolve foreign chain to foreign chain addresses (e.g., ETH->Solana)
2. Developers need to chain functions manually, increasing complexity and error risk
3. Unclear documentation leads to confusion when first encountering the system
4. Inconsistent error handling between the two functions

## Solution
This PR introduces a new universal resolution function that:
- Enables foreign chain to foreign chain resolution using NEAR as an intermediary
- Provides a single, clear interface for all token address resolution needs
- Maintains backwards compatibility with existing functions

## Key Changes
- Added `resolve_token_address` function that:
  - Handles all chain combinations through NEAR as intermediary
  - Provides consistent error handling via `Option<OmniAddress>` return type
  - Uses existing storage structures efficiently
- Added comprehensive test coverage
- Improved documentation explaining the intermediary system

## Technical Details
The resolution works by:
1. For NEAR source: directly looks up target chain address
2. For foreign chain source: 
   - First resolves to NEAR token ID
   - Then resolves from NEAR to target chain
3. Uses existing LookupMap infrastructure for O(1) lookups

## Testing
Test suite covers all resolution paths:
- NEAR → foreign chain
- Foreign chain → NEAR
- Foreign chain → foreign chain (via NEAR)
- Edge cases (unmapped tokens, same chain requests)
- Proper handling of chain-specific address formats (ETH, Solana)

## Migration
No migration needed as this is an additive change. Existing code paths remain unchanged.

## Security Considerations
- No changes to existing security model
- Maintains same validation checks as current implementation
- No new attack vectors introduced
- Uses same trusted storage and access patterns

## Performance Impact
- Minimal impact despite adding foreign-to-foreign resolution
- Maintains O(1) lookup complexity using existing storage structures
- Two lookups maximum for foreign-to-foreign resolution

### CODE REVIEW

File: near/omni-bridge/src/lib.rs [hunk 1]
```diff
@@ -892,6 +892,25 @@ impl Contract {
         }
     }
 
+    pub fn resolve_token_address(
+        &self,
+        token: &OmniAddress,
```
Comment (karim-en) [inline near/omni-bridge/src/lib.rs]:
```suggestion
        address: &OmniAddress,
```
The `token` and `token_id` are usually used for account id in our code.
For the OmniAddress, let's use `address`

File: near/omni-bridge/src/lib.rs [hunk 2]
```diff
@@ -892,6 +892,25 @@ impl Contract {
...
+        chain: ChainKind,
+    ) -> Option<OmniAddress> {
+        match token {
+            // If source is NEAR, directly look up in token_id_to_address
+            OmniAddress::Near(near_id) => self.token_id_to_address.get(&(chain, near_id.clone())),
+            // If source is foreign chain, first get NEAR id from token_address_to_id
+            foreign_addr => {
```
Comment (karim-en) [inline near/omni-bridge/src/lib.rs:900]:
```suggestion
            foreign_address => {
```

**Review by karim-en — CHANGES_REQUESTED** [review]

File: near/omni-bridge/src/lib.rs [hunk 3]
```diff
@@ -892,6 +892,35 @@ impl Contract {
         }
     }
 
+    pub fn get_bridged_token_address(
+        &self,
+        address: &OmniAddress,
```
Comment (frolvanya) [inline near/omni-bridge/src/lib.rs:897]: This should be named 'token' (at least in other methods we use it)

**Review by frolvanya — APPROVED** [review]

**Review by karim-en — APPROVED** [review]
